helix-term/tests/test/auto_pairs.rs

```diff
@@ -1,21 +1,543 @@
+use helix_core::{auto_pairs::DEFAULT_PAIRS, hashmap};
+
 use super::*;
 
+const LINE_END: &str = helix_core::DEFAULT_LINE_ENDING.as_str();
+
+fn differing_pairs() -> impl Iterator<Item = &'static (char, char)> {
+    DEFAULT_PAIRS.iter().filter(|(open, close)| open != close)
+}
+
+fn matching_pairs() -> impl Iterator<Item = &'static (char, char)> {
+    DEFAULT_PAIRS.iter().filter(|(open, close)| open == close)
+}
+
 #[tokio::test]
-async fn auto_pairs_basic() -> anyhow::Result<()> {
-    test(("#[\n|]#", "i(<esc>", "(#[|)]#\n")).await?;
+async fn insert_basic() -> anyhow::Result<()> {
+    for pair in DEFAULT_PAIRS {
+        test((
+            format!("#[{}|]#", LINE_END),
+            format!("i{}", pair.0),
+            format!("{}#[|{}]#{}", pair.0, pair.1, LINE_END),
+        ))
+        .await?;
+    }
 
-    test_with_config(
-        Args::default(),
-        Config {
-            editor: helix_view::editor::Config {
-                auto_pairs: AutoPairConfig::Enable(false),
-                ..Default::default()
-            },
+    Ok(())
+}
+
+#[tokio::test]
+async fn insert_configured_multi_byte_chars() -> anyhow::Result<()> {
+    // [NOTE] these are multi-byte Unicode characters
```

Comment:
tree-sitter-comment would make this a little more high-profile as `NOTE:` rather than `[NOTE]`